src/audio/kpb.c

```diff
@@ -315,7 +315,7 @@ static void kpb_free_history_buffer(struct hb *buff)
 		_buff = buff->next;
 		rfree(buff);
 		buff = _buff;
-	} while (buff != first_buff);
+	} while (buff && buff != first_buff);
```

Comment:
This is unrelated to the heap buffer increase I think, so it should be a separate commit (if it's even needed)

Reply:
@paulstelian97 It is a separate commit. And it's needed to not get panic in case buffer is too small.

Reply:
For some reason it didn't show up as separate for me when I reviewed?

Reply:
@tlauda, but the history buffer allocator guarantees 'buff->next' is never NULL so the problem you are trying to fix should theoretically never happen, does it?

Reply:
@mrajwa But it happens. Just don't increase the heap size on JSL.

Reply:
@tlauda  Why JSL ?

Reply:
@lgirdwood I bet it's generic issue for every platform. I've mentioned JSL, because it's the only platform currently having too less memory to allocate the history buffer. I guess @mrajwa is right and there is another bug in kpb allocation buffer procedure, which leaves one of the items in the linked list NULL. Nevertheless this fix is also needed to make sure we're going to always gracefully recover.

Reply:
Is this something we can pick up with cmocka UT testing ?

Reply:
@lgirdwood, well it only happens when memory allocation fails and we don't have a UT which mimics such case.
@lgirdwood, I know the root cause of it, PR is comming soon.